src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobScheduler.scala

```diff
@@ -507,21 +507,24 @@ class JobScheduler @Inject()(val scheduleHorizon: Period,
           (None, None)
         } else {
           val job = jobOption.get
-          if (nextDate.isAfter(now.minus(job.epsilon)) && nextDate.isBefore(now.plus(scheduleHorizon))) {
+          val scheduleWindowBegin = now.minus(job.epsilon)
+          val scheduleWindowEnd = now.plus(scheduleHorizon)
+          if (nextDate.isAfter(scheduleWindowBegin) && nextDate.isBefore(scheduleWindowEnd)) {
             log.info("Task ready for scheduling: %s".format(nextDate))
             //TODO(FL): Rethink passing the dispatch queue all the way down to the ScheduledTask.
             val task = new ScheduledTask(TaskUtils.getTaskId(job, nextDate), nextDate, job, taskManager)
-            return (Some(task), stream.tail())
+            return (Some(task), stream.tail)
           }
-          //The nextDate has passed already beyond epsilon.
-          //TODO(FL): Think about the semantics here and see if it always makes sense to skip ahead of missed schedules.
+          // Next instance is too far in the future
+          // Needs to be scheduled at a later time, after schedule horizon.
           if (!nextDate.isBefore(now)) {
-            jobsObserver.apply(JobSkipped(job, nextDate))
             return (None, Some(stream))
           }
-          //Needs to be scheduled at a later time, after schedule horizon.
+          // Next instance is too far in the past (beyond epsilon)
+          //TODO(FL): Think about the semantics here and see if it always makes sense to skip ahead of missed schedules.
           log.fine("No need to work on schedule: '%s' yet".format(nextDate))
```

Comment:
If you follow the logic, the first check on line 512 checks if he next instance is in the scheduling window, if so, return it.
The next check was to see if the remaining instances are too far in the future, if so, nothing to add to the queue, but nothing is skipped either.
We get here IFF the instance we are looking at is from the past, from _before_ the scheduling window. When that happens, notice that we just drop the head and make a recursive call via next with remaining scheduled times.

So, I argue that this call to skipped is now in the correct place, while before it was not